### PR TITLE
Rewrite test script.

### PR DESCRIPTION
**Pull request by rabbah**
This gives greater transparency into why tests fail.
pg1/785.

### CODE REVIEW

File: tests/src/whisk/core/database/test/CacheConcurrencyTests.scala [hunk 1]
```diff
@@ -31,45 +29,112 @@ import common.Wsk
...
-    for (i <- 1 to nExternalIters)
-        "the cache" should s"support concurrent CRUD without bogus residual cache entries, iter ${i}" in {
+    def run[W](phase: String)(block: String => W) = parallel.map { i =>
+        val name = s"testy${i}"
+        withClue(s"$phase: failed for $name") { (name, block(name)) }
+    }
+
+    for (n <- 1 to nExternalIters)
+        "the cache" should s"support concurrent CRUD without bogus residual cache entries, iter ${n}" in {
             try {
```
Comment (markusthoemmes) [inline tests/src/whisk/core/database/test/CacheConcurrencyTests.scala]: Does it make sense to drop the `try` and implement the cleanup in an `afterEach`?

File: tests/src/whisk/core/database/test/CacheConcurrencyTests.scala [hunk 2]
```diff
@@ -31,45 +29,112 @@ import common.Wsk
...
                 }
 
-                exitCode should be(0)
+                run("delete+get") {
+                    name =>
+                        val para = (1 to 30).par
+                        para.tasksupport = new ForkJoinTaskSupport(new ForkJoinPool(nThreads))
+                        para.map {
+                            i =>
+                                if (i != 16) {
```
Comment (markusthoemmes) [inline tests/src/whisk/core/database/test/CacheConcurrencyTests.scala]: What's that magic number? Maybe add a comment or give it a name?

Comment (rabbah) [inline tests/src/whisk/core/database/test/CacheConcurrencyTests.scala]: half way point, nothing special. will add comment.

**Comment by rabbah** [comment]
@starpit please review the test changes. @markusthoemmes deferring to you to merge; comments addressed and pg tag pending.

**Comment by starpit** [comment]
i think this is an accurate port from the hellish bash script. should the PR include eliminating bash script, or that script still useful for standalone testing?

**Comment by markusthoemmes** [comment]
rebased and merged in 8d747fa1d4e1cfee496cfe6f568876389487e372

**Pull request closed without merging**